### PR TITLE
Processing center visit 

### PR DESCRIPTION
**Pull request by nishant22029**
- Added changes to rediect user after Processing center QR code scan to visit purpose page
- Added changes to redirect user to check user exist or not for Office Visit, and material contribution button clicked
Screenshot 2024-09-02 at 2 03 22 PM

### CODE REVIEW

File: wp-content/themes/goonj-crm/functions.php [hunk 1]
```diff
@@ -236,6 +236,17 @@ function goonj_handle_user_identification_form() {
 			exit;
 		}
 
+		if ( 'office-visit-material-contribution' === $purpose ) {
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:239]: Aren't office visit and material contribution two different purpose?

I was expecting something like: `processing-center-material-contribution` and `processing-center-office-visit`

File: wp-content/themes/goonj-crm/functions.php [hunk 2]
```diff
@@ -236,6 +236,17 @@ function goonj_handle_user_identification_form() {
...
+			$institute_registration_form_path = sprintf(
+				'/office-visit/material-contribution/#?email=%s&phone=%s&target_id=%s',
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:241]: Are we doing material contribution inside office visit or are we doing material contribution inside processing center?

```suggestion
				'/office-visit/material-contribution/#?email=%s&phone=%s&target_id=%s',
```

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/themes/goonj-crm/functions.php (hunk 2 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: We are doing inside Processing center

File: wp-content/themes/goonj-crm/functions.php (hunk 1 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: sure updated

File: wp-content/plugins/goonj-blocks/src/render.php [hunk 3]
```diff
@@ -24,6 +24,15 @@
 	$action_target['id'],
 );
 
+$pu_visit_check_link = sprintf(
+	'/office-visit/?target_id=%s',
```
Comment (pokhiii) [inline wp-content/plugins/goonj-blocks/src/render.php:28]: I don't undertand this link structure. All these actions are happening inside processing center so why are we not doing:
```suggestion
	'/processing-center/office-visit/?target_id=%s',
```

File: wp-content/plugins/goonj-blocks/src/render.php [hunk 4]
```diff
@@ -24,6 +24,15 @@
...
+	$action_target['id']
+);
+
+$pu_material_contribution_check_link = sprintf(
+	'/processing-center-material-contribution-check-user/?target_id=%s',
```
Comment (pokhiii) [inline wp-content/plugins/goonj-blocks/src/render.php:33]:
```suggestion
	'/processing-center/material-contribution/?target_id=%s',
```

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/plugins/goonj-blocks/src/render.php (hunk 3 above)
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/render.php:28]: I previously attempted to create the URL `processing-center/office-visit`, but when updating the URL in the pages, it automatically changes to `processing-center-office-visit`, preventing the desired format.

File: wp-content/plugins/goonj-blocks/src/render.php (hunk 4 above)
Comment (nishant22029) [reply wp-content/plugins/goonj-blocks/src/render.php]: updated urls

**Review by pokhiii — APPROVED** [review]

**Pull request merged**